config/config.go

```diff
@@ -72,6 +75,20 @@ func ReadConfig() (ShipperConfig, error) {
 	return config, nil
 }
 
+// ShipperConfigFromUnitConfig converts the configuration provided by Agent to the internal
+// configuration object used by the shipper.
+// Currently this just converts the given struct to json and tries to deserialize it into the
+// ShipperConfig struct. This is not the right way to do this, but this gets the build and
+// tests passing again with the new version of elastic-agent-client. Migrating fully to this
+// new config structure is part of the overall agent V2 transition.
```

Comment:
You can reference this issue to define the final format of the configuration if you want: https://github.com/elastic/elastic-agent/issues/617